### PR TITLE
feat: Remove username from user invitation link

### PR DESCRIPTION
**Pull request by netroms**
Remove username from invitation and restore URL, use a unique token instead.

Issue: [DHIS2-8096]

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserService.java [hunk 1]
```diff
@@ -1,392 +1,400 @@
...
+     */
+    public UserCredentials getUserCredentialsBySecret( String token );
+
+    /**
+     * Retrieves the UserCredentials associated with the User with the given
+     * name.
+     *
+     * @param username the name of the User.
+     * @return the UserCredentials.
+     */
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserService.java:216]: Do you know why we get a complete diff on this file, line ending issue?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserCredentialsStore.java [hunk 2]
```diff
@@ -73,4 +73,12 @@ public UserCredentials getUserCredentialsByLdapId( String ldapId )
         query.setParameter( "ldapId", ldapId );
         return query.uniqueResult();
     }
+
+    @Override
+    public UserCredentials getUserCredentialsBySecret( String secret )
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserCredentialsStore.java:78]: Let's introduce a new property for this we use only for invites/password reset. The secret field is used as a shared secret for 2FA and we should be very careful not expose it through the API at all.

File: dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/mock/MockUserService.java [hunk 3]
```diff
@@ -166,6 +166,19 @@ public void updateUserCredentials( UserCredentials userCredentials )
     {
     }
 
+    @Override
+    public UserCredentials getUserCredentialsBySecret( String secret )
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/mock/MockUserService.java:170]: See common on new property.

**Review by stian-sandvold — APPROVED** [review]
Looks good to me, pending @larshelge 's comments.

Is this ready to be converted to a normal PR?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserCredentialsStore.java (hunk 2 above)
Comment (netroms) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserCredentialsStore.java]: New property made, the name is "idtoken", I also removed an old property not in use anymore called "restorecode"

**Review by larshelge — APPROVED** [review]

**Pull request merged**